core/core-impl/src/main/java/io/openshift/appdev/missioncontrol/tracking/segment/SegmentAnalyticsProvider.java

```diff
@@ -0,0 +1,102 @@
+package io.openshift.appdev.missioncontrol.tracking.segment;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.logging.Logger;
+
+import javax.annotation.Resource;
+import javax.enterprise.concurrent.ManagedExecutorService;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Produces;
+
+import com.segment.analytics.Analytics;
+import com.segment.analytics.messages.MessageBuilder;
+import com.segment.analytics.messages.TrackMessage;
+
+import io.openshift.appdev.missioncontrol.core.api.LaunchEvent;
+import io.openshift.appdev.missioncontrol.core.api.StatusMessage;
+import io.openshift.appdev.missioncontrol.core.api.StatusMessageEvent;
+
+/**
+ * Provides the {@link Analytics} instance to be used
+ *
+ * @author <a href="mailto:alr@redhat.com">Andrew Lee Rubinger</a>
+ * @author <a href="mailto:tschotan@redhat.com">Tako Schotanus</a>
+ */
+@ApplicationScoped
+public class SegmentAnalyticsProvider {
+
+    private static final Logger log = Logger.getLogger(SegmentAnalyticsProvider.class.getName());
+
+    private static final String NAME_EVENT_LAUNCH = "launch";
+    private static final String KEY_OPENSHIFT_PROJECT_NAME = "openshiftProjectName";
+    private static final String KEY_GITHUB_REPO = "githubRepo";
+
+    //TODO Externalize into sysprop
+    private static final String KEY = "oOlNiAf3K5MDwpd4ErD2ZPRe6z3Ckk7w";
+
+    private Analytics analytics;
+
+    @Resource
+    private ManagedExecutorService async;
+
+    public void initAnalytics(@Observes @Initialized(ApplicationScoped.class) Object init) {
```

Comment:
Does this need to be initialized prematurely? How about using @PostConstruct instead?